staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go

```diff
@@ -30,12 +30,39 @@ import (
 	"google.golang.org/grpc"
 
 	kmsapi "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v1beta1"
+	"sync"
+	"time"
 )
 
 const (
 	endpoint = "unix:///@kms-socket.sock"
 )
 
+// TestKMSPluginLateStart tests the scenario where kms-plugin pod/container starts after kube-apiserver pod/container.
+func TestKMSPluginLateStart(t *testing.T) {
+	var wg sync.WaitGroup
```

Comment:
Hmm, I think you want the opposite: kick off goroutine that sleeps for 5s and then starts KMS provider and in main test body call NewGRPCService. That way NewGRPCService should be waiting until KMS provider is started.